src/main/java/net/minecraftforge/common/Tags.java

```diff
@@ -5,223 +5,591 @@
+         */
+        public static final TagKey<EntityType<?>> CAPTURING_NOT_SUPPORTED = cTag("capturing_not_supported");
+
+        /**
+         * Entities should be included in this tag if they are not allowed to be teleported in any way.
+         * This is more for mods that allow teleporting entities within the same dimension. Any mod that is
+         * teleporting entities to new dimensions should be checking canChangeDimensions method on the entity itself.
+         */
+        public static final TagKey<EntityType<?>> TELEPORTING_NOT_SUPPORTED = cTag("teleporting_not_supported");
+        //endregion
 
-        private static TagKey<EntityType<?>> tag(String name)
-        {
-            return TagKey.create(Registries.ENTITY_TYPE, ResourceLocation.fromNamespaceAndPath("forge", name));
+        private static TagKey<EntityType<?>> cTag(String name) {
+            return TagKey.create(Registries.ENTITY_TYPE, ResourceLocation.fromNamespaceAndPath("c", name));
         }
     }
 
-    public static class Items
-    {
-        private static void init(){}
-
-        public static final TagKey<Item> BARRELS = tag("barrels");
-        public static final TagKey<Item> BARRELS_WOODEN = tag("barrels/wooden");
-        public static final TagKey<Item> BONES = tag("bones");
-        public static final TagKey<Item> BOOKSHELVES = tag("bookshelves");
-        public static final TagKey<Item> CHESTS = tag("chests");
-        public static final TagKey<Item> CHESTS_ENDER = tag("chests/ender");
-        public static final TagKey<Item> CHESTS_TRAPPED = tag("chests/trapped");
-        public static final TagKey<Item> CHESTS_WOODEN = tag("chests/wooden");
-        public static final TagKey<Item> COBBLESTONE = tag("cobblestone");
-        public static final TagKey<Item> COBBLESTONE_NORMAL = tag("cobblestone/normal");
-        public static final TagKey<Item> COBBLESTONE_INFESTED = tag("cobblestone/infested");
-        public static final TagKey<Item> COBBLESTONE_MOSSY = tag("cobblestone/mossy");
-        public static final TagKey<Item> COBBLESTONE_DEEPSLATE = tag("cobblestone/deepslate");
-        public static final TagKey<Item> CROPS = tag("crops");
-        public static final TagKey<Item> CROPS_BEETROOT = tag("crops/beetroot");
-        public static final TagKey<Item> CROPS_CARROT = tag("crops/carrot");
-        public static final TagKey<Item> CROPS_NETHER_WART = tag("crops/nether_wart");
-        public static final TagKey<Item> CROPS_POTATO = tag("crops/potato");
-        public static final TagKey<Item> CROPS_WHEAT = tag("crops/wheat");
-        public static final TagKey<Item> DUSTS = tag("dusts");
-        public static final TagKey<Item> DUSTS_PRISMARINE = tag("dusts/prismarine");
-        public static final TagKey<Item> DUSTS_REDSTONE = tag("dusts/redstone");
-        public static final TagKey<Item> DUSTS_GLOWSTONE = tag("dusts/glowstone");
-
-        public static final TagKey<Item> DYES = tag("dyes");
+    public static class Items {
```

Comment:
You are missing the following item tags:

- `c:dyes/black`
  - `minecraft:black_dye`
- `c:dyes/blue`
  - `minecraft:blue_dye`
- `c:dyes/brown`
  - `minecraft:brown_dye`
- `c:dyes/cyan`
  - `minecraft:cyan_dye`
- `c:dyes/gray`
  - `minecraft:gray_dye`
- `c:dyes/green`
  - `minecraft:green_dye`
- `c:dyes/light_blue`
  - `minecraft:light_blue_dye`
- `c:dyes/light_gray`
  - `minecraft:light_gray_dye`
- `c:dyes/lime`
  - `minecraft:lime_dye`
- `c:dyes/magenta`
  - `minecraft:magenta_dye`
- `c:dyes/orange`
  - `minecraft:orange_dye`
- `c:dyes/pink`
  - `minecraft:pink_dye`
- `c:dyes/purple`
  - `minecraft:purple_dye`
- `c:dyes/red`
  - `minecraft:red_dye`
- `c:dyes/white`
  - `minecraft:white_dye`
- `c:dyes/yellow`
  - `minecraft:yellow_dye`